resources.d.lua

```diff
@@ -9,28 +9,15 @@
 ---@module 'math'
 
 ---@class Resources
----@field zones ResourceZoneInfo[] List of known resource zones
----@field [integer]ResourceZoneInfo Known resource zones
```

Comment:
this split of `[integer]ResourceZoneInfo` and `zones ResourceZoneInfo[]`  was meant as giving both a way to get them as an ordered list and directly